internal/storage/v1/elasticsearch/spanstore/reader.go

```diff
@@ -452,9 +452,9 @@ func (s *SpanReader) multiRead(ctx context.Context, traceIDs []dbmodel.TraceID,
 		}
 	}
 
-	var traces []*dbmodel.Trace
+	var traces []dbmodel.Trace
```

Comment:
this should be done differently
  * reserve `traces` slice upfront with the capacity of `len(traceIDs)`. Can do it before L376
  * in L444 (when trace object not found in the tracesMap), append it to `traces` output slice, and store in the map as `tracesMap[lastSpan.TraceID] = &traces[len(traces)-1]`. This way you don't allocate memory for each Trace object, only once for all of them in the slice
  * this loop is not needed anymore